src/tools/todo.zig

```diff
@@ -11,7 +11,19 @@ pub fn work(alloc: std.mem.Allocator, file_name: []const u8, src: *main.Source,
     while (iter.next()) |line| : (i += 1) {
         const lll = std.mem.trimLeft(u8, line, " ");
         const col = line.len - lll.len + 1;
-        if (!std.mem.startsWith(u8, lll, "// TODO")) continue;
-        try writer.print("./{s}:{d}:{d}: TODO: {s}\n", .{ file_name, i, col, std.mem.trim(u8, lll[7..], " ") });
+        if (lll.len < 7 or !std.mem.startsWith(u8, lll, "//")) continue;
+        // allow for //, ///, or //! comments
+        switch (lll[2]) {
+            '/', '!', ' ' => {},
+            else => continue,
+        }
+        const comment_str = std.mem.trimLeft(u8, lll[3..], " ");
+        if (!std.mem.startsWith(u8, comment_str, "TODO")) continue;
+        const todo_msg = msg: {
+            // skip colon character if it's immediately after TODO
+            const msg_start: usize = if (std.mem.startsWith(u8, comment_str[4..], ":")) 5 else 4;
+            break :msg std.mem.trim(u8, comment_str[msg_start..], " ");
+        };
```

Comment:
can this be replaced with `std.mem.trimLeft(u8, msg, ": ");`

Reply:
I intentionally avoided that. From the OP:

> Note: The logic for skipping `:` is a little verbose to avoid the (admittedly unlikely) situation where it could trim too many colons from something like `// TODO ::SomeWeirdThing`